### PR TITLE
[Replica][Config] Add max/min replica config for create/alter table

### LINKED ISSUE #6035: [Feature] Add min/max replication number limit for create or alter table
**Is your feature request related to a problem? Please describe.**
We found some user create a table with only one replica. It is high risking for losing data. We also found some users own a table with more than 5 replicas, it is a big waste of storage.

**Describe the solution you'd like**
Add 2 configurations: 
`public static int min_replica_num = 1;`
`public static int max_replica_num = Integer.MAX_VALUE;`

When user create a table or alter a table, it will check whether the property `replication_num` and `dynamic_partition.replication_num`  is in [min_replica_num, min_replica_num]

It limits the replication number of a table, but also a partition.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by xinghuayu007**
## Proposed changes

We found some user create a tale with only one replica. It is high risking for losing data. We also found some users own a table with more than 5 replicas, it is big waste of storage.

Therefore this PR add min/max replica config to limit creating/altering table.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #6035) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If these changes need document changes, I have updated the document
- [x] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -1637,6 +1637,11 @@ public void process(List<AlterClause> alterClauses, String clusterName, Database
                         return;
                     } else if (properties.containsKey("default." + PropertyAnalyzer.PROPERTIES_REPLICATION_NUM)) {
                         Preconditions.checkNotNull(properties.get(PropertyAnalyzer.PROPERTIES_REPLICATION_NUM));
+                        short replicaNum = Short.parseShort(properties.getOrDefault(PropertyAnalyzer.PROPERTIES_REPLICATION_NUM,
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeHandler.java:1640]: The maximum number of copies should be the number of be not be the max Integer

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java [hunk 2]
```diff
@@ -3631,6 +3631,9 @@ private void createOlapTable(Database db, CreateTableStmt stmt) throws DdlExcept
             boolean isReplicationNumSet = properties != null && properties.containsKey(PropertyAnalyzer.PROPERTIES_REPLICATION_NUM);
             replicationNum = PropertyAnalyzer.analyzeReplicationNum(properties, replicationNum);
             if (isReplicationNumSet) {
+                if (replicationNum < Config.min_replica_num || replicationNum > Config.max_replica_num) {
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Catalog.java:3634]: same as above.

File: fe/fe-core/src/main/java/org/apache/doris/common/Config.java [hunk 3]
```diff
@@ -1396,4 +1396,12 @@
      */
     @ConfField(mutable = true, masterOnly = true)
     public static int max_dynamic_partition_num = 500;
+
+    /**
+     * min/max replica number limit
+     */
+    @ConfField(mutable = true)
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/common/Config.java:1403]: The minimum and maximum values should not be a configuration, but a hard-coded constant.

File: fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java [hunk 4]
```diff
@@ -187,8 +187,9 @@ private static void checkReplicationNum(String val) throws DdlException {
             throw new DdlException("Invalid properties: " + DynamicPartitionProperty.REPLICATION_NUM);
         }
         try {
-            if (Integer.parseInt(val) <= 0) {
-                ErrorReport.reportDdlException(ErrorCode.ERROR_DYNAMIC_PARTITION_REPLICATION_NUM_ZERO);
+            if (Integer.parseInt(val) < Config.min_replica_num || Integer.parseInt(val) > Config.max_replica_num) {
```
Comment (EmmyMiao87) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/DynamicPartitionUtil.java:190]: same as above

**Comment by xinghuayu007** [comment]
I think this feature is better to be done in Data Management System, not in Doris.

**Comment by EmmyMiao87** [comment]
> I think this feature is better to be done in Data Management System, not in Doris.

I will close this PR~

**Pull request closed without merging**